src/dense/generic_dense.jl

```diff
@@ -313,6 +313,26 @@ function evaluate_interpolant(f, Θ, dt, timeseries, i₋, i₊, cache, idxs,
     end
 end
 
+function get_differential_vars(f, idxs, size)
+    differential_vars = nothing
+    if hasproperty(f, :mass_matrix)
+        mm = f.mass_matrix
+        if mm isa UniformScaling
+            # already correct
+        elseif isdiag(mm)
+            differential_vars = reshape(Diagonal(mm).diag .!= 0, size)
```

Comment:
Isn't this the same as:
```suggestion
            differential_vars = reshape(diag(mm)  .!= 0, size)
```

Reply:
Seems like it